### PR TITLE
Moved UserMode from PortalSettings to Personalization class

### LINKED ISSUE #4213: IPortalSettings lacking UserMode
While trying to resolve build warnings on DotNetNuke.Common.Globals, I found the method IsEditMode() had a warning about using the deprecated PortalController.Instance.GetCurrentPortalSettings and replaced it with .GetCurrentSettings which returns an IPortalSettings instead of a PortalSettings object.

IPortalSettings is lacking a UserMode property that PortalSettings had.

Should we :
🅰️ Add UserMode to the interface ?
🅱️ Deprecated UserMode in PortalSettings with some suggestion on how to do it the new way ?

Paging @bdukes @ahoefling

### PR DESCRIPTION
**Pull request by valadas**
Closes #4213

I am open to any feedback on this, I do believe this way would not be a breaking change...

### CODE REVIEW

File: DNN Platform/Library/Services/Personalization/Personalization.cs [hunk 1]
```diff
@@ -147,6 +152,38 @@ public static void SetSecureProfile(PersonalizationInfo personalization, string
...
+        }
+
+        /// <summary>
+        /// Gets the mode the user is viewing the page in.
+        /// </summary>
+        /// <returns><see cref="Mode"/>.</returns>
+        public static Mode GetUserMode()
+        {
+            Mode mode;
+            if (HttpContext.Current != null && HttpContext.Current.Request.IsAuthenticated)
```
Comment (bdukes) [inline DNN Platform/Library/Services/Personalization/Personalization.cs:164]:
```suggestion
            if (HttpContextSource.Current?.Request.IsAuthenticated == true)
```

File: DNN Platform/Library/Services/Personalization/Personalization.cs [hunk 2]
```diff
@@ -147,6 +152,38 @@ public static void SetSecureProfile(PersonalizationInfo personalization, string
...
+            {
+                mode = PortalSettings.Current.DefaultControlPanelMode;
+                string setting = Convert.ToString(Personalization.GetProfile("Usability", "UserMode" + PortalController.Instance.GetCurrentSettings().PortalId));
+                switch (setting.ToUpper())
```
Comment (bdukes) [inline DNN Platform/Library/Services/Personalization/Personalization.cs:168]:
```suggestion
                switch (setting.ToUpperInvariant())
```

File: DNN Platform/Library/Services/Personalization/Personalization.cs [hunk 3]
```diff
@@ -2,22 +2,27 @@
...
-{
+{
+    using System;
     using System.Web;
 
     using DotNetNuke.Common.Utilities;
     using DotNetNuke.Entities.Portals;
     using DotNetNuke.Entities.Users;
     using DotNetNuke.Security;
+    using static DotNetNuke.Entities.Portals.PortalSettings;
```
Comment (bdukes) [inline DNN Platform/Library/Services/Personalization/Personalization.cs:13]: Doesn't look like this is in use
```suggestion
    using static DotNetNuke.Entities.Portals.PortalSettings;
```

File: DNN Platform/Library/Entities/Portals/PortalSettings.cs [hunk 4]
```diff
@@ -201,6 +215,10 @@ public UserInfo UserInfo
             }
         }
 
+        /// <summary>
+        /// Gets the mode the user is viewing the page in.
+        /// </summary>
+        [Obsolete("Deprecated in v9.8.1, use Personalization.GetUserMode() instead, Scheduled for removal in v10.")]
         public Mode UserMode
         {
             get
```
Comment (bdukes) [inline DNN Platform/Library/Entities/Portals/PortalSettings.cs:224]: Should the body of this property just return `Personalization.GetUserMode()`, so we're not duplicating the implementation?

**Review by bdukes — APPROVED** [review]
LGTM, thanks!

File: DNN Platform/Library/Services/Personalization/Personalization.cs (hunk 3 above)
Comment (valadas) [reply DNN Platform/Library/Services/Personalization/Personalization.cs:13]: Yes it is used for `PortalSettings.Current.DefaultControlPanelMode`

File: DNN Platform/Library/Entities/Portals/PortalSettings.cs (hunk 4 above)
Comment (valadas) [reply DNN Platform/Library/Entities/Portals/PortalSettings.cs]: Yep, forgot to do that, it is pushed in my latest commit. Thanks.

**Review by david-poindexter — APPROVED** [review]

**Pull request merged**